core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -37,11 +37,15 @@
     {
         super(rowIndex);
         this.fieldBlockBuilders = fieldBlockBuilders;
```

Comment:
do not compute `initialBlockBuilderSize` eagerly in constructor or in `reset`. I don't even know if `SingleRowBlockWriter#getSizeInBytes` is really used in practice.

BTW: this might be a reason why you don't see much improvement in benchmarks

Reply:
ok :)

Reply:
ok